packages/api/tests/animalTask.test.js

```diff
@@ -840,11 +829,26 @@ describe('Animal task tests', () => {
           return completeTaskRequest({ user_id, farm_id }, data, task_id, 'animal_movement_task');
         };
 
-        const checkAnimalMovement = (original, completedTask) => {
+        const checkAnimalMovement = async (original, completedTask) => {
           expect(completedTask.animals.length).toBe(original.animals.length);
           expect(completedTask.animal_batches.length).toBe(original.animal_batches.length);
-          [...completedTask.animals, ...completedTask.animal_batches].forEach((animalOrBatch) => {
-            expect(animalOrBatch.location_id).toBe(location_id);
+          const animalsData = await knex('animal')
+            .select('location_id')
+            .whereIn(
+              'id',
+              completedTask.animals.map(({ id }) => id),
+            );
+          const batchesData = await knex('animal_batch')
+            .select('location_id')
+            .whereIn(
+              'id',
+              completedTask.animal_batches.map(({ id }) => id),
+            );
+          expect(completedTask.animals.length + completedTask.animal_batches.length).toBe(
+            animalsData.length + batchesData.length,
+          );
+          [...animalsData, ...batchesData].forEach((data) => {
+            expect(data.location_id).toBe(location_id);
```

Comment:
The API response no longer includes the animals' location, so the location in the DB is now being checked.